main_test.go

```diff
@@ -1,17 +1,56 @@
 package main
 
-import "testing"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
 
-func TestValidateInstance(t *testing.T) {
-	resource = "myinstance"
-	fakeInstance := ResourceInstance{Resource: "myinstance", Id: "i-12356"}
-	fakeMap := map[string]ResourceInstance{"myinstance": fakeInstance}
-	fakeResult := TerraformInstanceResult{Instances: fakeMap, Error: nil}
+func openAndUnmarshal(i interface{}, path string) error {
+	f, err := os.Open(path)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	defer f.Close()
+
+	byteValue, _ := ioutil.ReadAll(f)
+
+	err = json.Unmarshal([]byte(byteValue), &i)
```

Comment:
If there are only a couple of types to be unmarshalled, it may be worth writing dedicated type-safe methods for the job.